packages/lib-ts/src/intents/Transfer.ts

```diff
@@ -28,30 +30,38 @@ export class TransferBuilder extends IntentBuilder {
   }
 
   addTransfers(transfers: TransferData[]): TransferBuilder {
-    for (let i = 0; i < transfers.length; i++) {
-      this.transfers.push(transfers[i])
-    }
+    for (let i = 0; i < transfers.length; i++) this.transfers.push(transfers[i])
     return this
   }
 
   addTransferFromTokenAmount(tokenAmount: TokenAmount, recipient: Address): TransferBuilder {
-    if (tokenAmount.token.chainId !== this.chainId) {
-      throw new Error('All tokens must be on the same chain')
-    }
+    if (tokenAmount.token.chainId !== this.chainId) throw new Error('Transfer tokens must be on the same chain')
     return this.addTransfer(TransferData.fromTokenAmount(tokenAmount, recipient))
   }
 
+  addTransferFromI32(token: Token, amount: i32, recipient: Address): TransferBuilder {
+    if (token.chainId !== this.chainId) throw new Error('Transfer tokens must be on the same chain')
+    return this.addTransfer(TransferData.fromI32(token, amount, recipient))
+  }
+
+  addTransferFromBigInt(token: Token, amount: i32, recipient: Address): TransferBuilder {
+    if (token.chainId !== this.chainId) throw new Error('Transfer tokens must be on the same chain')
+    return this.addTransfer(TransferData.fromI32(token, amount, recipient))
+  }
```

Comment:
Shouldn't this be...
```suggestion
  addTransferFromBigInt(token: Token, amount: BigInt, recipient: Address): TransferBuilder {
    if (token.chainId !== this.chainId) throw new Error('Transfer tokens must be on the same chain')
    return this.addTransfer(TransferData.fromBigInt(token, amount, recipient))
  }
```
?